src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machines = new ArrayList<>();
+
+        if (type == Bulldozer.class) {
+            machines.addAll(new BulldozerProducer().get());
+            return machines;
```

Comment:
Don't return null or specific type List in `getAll()` when no producer has been found. Instead of returning a new ArrayList at the end of the method, you could return the `machines` list directly, which will be empty if none of the conditions are met.